### ISSUE TITLE
bitbake xcsoar-image-testing

### ISSUE THREAD

**Issue by lordfolken** [Openvario/meta-openvario#144]
With hardknott xcsoar-image-testing fails to compile sensord-testing.

xcsoar-image btibake target is fine. 

```bash
Build Configuration:
BB_VERSION           = "1.50.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "arm-ovlinux-linux-gnueabi"
MACHINE              = "openvario-57-lvds"
DISTRO               = "ovlinux"
DISTRO_VERSION       = "21343"
TUNE_FEATURES        = "arm armv7ve vfp thumb neon callconvention-hard"
TARGET_FPU           = "hard"
meta                 
meta-poky            
meta-yocto-bsp       = "HEAD:7edd2be25fdf72187738fc551235df0f24e918bb"
meta-sunxi           = "HEAD:612de90a00fbd89762e3d85d85c6846fdc1f9ef6"
meta-openvario       = "HEAD:7c349522ada91ccae88766593fe3da4af2b22b3e"
meta-oe              
meta-python          
meta-networking      = "HEAD:f72a73b42fa740130b388ba8555cdbefdee8d37d"


NOTE: Executing Tasks
ERROR: sensord-testing-git-r10 do_compile: ExecutionError('/workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/run.do_compile.146', 2, None, None)
ERROR: Logfile of failure stored in: /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/log.do_compile.146
Log data follows:
| DEBUG: Executing shell function do_compile
| Making ..
| /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10
| arm-ovlinux-linux-gnueabi-gcc  -march=armv7ve -mthumb -mfpu=neon -mfloat-abi=hard -fstack-protector-strong  -O2 -D_FORTIFY_SOURCE=2 -Wformat -Wformat-security -Werror=format-security --sysroot=/workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/recipe-sysroot -g -o sensord obj/ms5611.o obj/ams5915.o obj/ads1110.o obj/main.o obj/nmea.o obj/timer.o obj/KalmanFilter1d.o obj/cmdline_parser.o obj/configfile_parser.o obj/vario.o obj/AirDensity.o obj/24c16.o obj/ds2482.o obj/humidity.o -lrt -lm
| /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/recipe-sysroot-native/usr/bin/arm-ovlinux-linux-gnueabi/../../libexec/arm-ovlinux-linux-gnueabi/gcc/arm-ovlinux-linux-gnueabi/10.2.0/ld: obj/main.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: multiple definition of `sensor_prev'; obj/ms5611.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: first defined here
| /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/recipe-sysroot-native/usr/bin/arm-ovlinux-linux-gnueabi/../../libexec/arm-ovlinux-linux-gnueabi/gcc/arm-ovlinux-linux-gnueabi/10.2.0/ld: obj/configfile_parser.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: multiple definition of `sensor_prev'; obj/ms5611.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: first defined here
| /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/recipe-sysroot-native/usr/bin/arm-ovlinux-linux-gnueabi/../../libexec/arm-ovlinux-linux-gnueabi/gcc/arm-ovlinux-linux-gnueabi/10.2.0/ld: obj/24c16.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: multiple definition of `sensor_prev'; obj/ms5611.o:/usr/src/debug/sensord-testing/git-r10/git/ms5611.h:59: first defined here
| collect2: error: ld returned 1 exit status
| make: *** [Makefile:32: sensord] Error 1
| WARNING: /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/run.do_compile.146:144 exit 2 from 'make'
| WARNING: Backtrace (BB generated script):
| 	#1: do_compile, /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/run.do_compile.146, line 144
| 	#2: main, /workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/run.do_compile.146, line 148
| ERROR: ExecutionError('/workdir/poky/build/tmp/work/armv7vet2hf-neon-ovlinux-linux-gnueabi/sensord-testing/git-r10/temp/run.do_compile.146', 2, None, None)
ERROR: Task (/workdir/poky/meta-openvario/recipes-apps/sensord/sensord-testing_git.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 3951 tasks of which 3948 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /workdir/poky/meta-openvario/recipes-apps/sensord/sensord-testing_git.bb:do_compile
Summary: There was 1 ERROR message shown, returning a non-zero exit code.
```

**Comment by August2111** [comment]
For this sensord-testing issue I did a PR on the sensord repository yesterday.
I hope this helps...
But after this you are running in a compiling error with xcsoar-testing, this has to do with the new code prepared for 7.21...
Maybe we should use the for this testing image the regular published xcsoar: The changed code here is not official published, and the published version is not far away from this (The big difference between this two setups between 2017 and beginning of this year is disappeared starting with XCSoar 7.1)

**Comment by linuxianer99** [comment]
Fixed by openvario/sensord#35
